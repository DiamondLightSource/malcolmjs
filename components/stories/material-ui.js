import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import AppBar from 'material-ui/AppBar';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/RaisedButton';
import Toggle from 'material-ui/Toggle';
import { storiesOf, action } from '@kadira/storybook';

function handleEdgeDeleteButton(){

  console.log("material-ui button clicked");

}

storiesOf('Button', module)
    .add('material-ui flat button', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(lightBaseTheme)}>

            <FlatButton key="edgeDeleteButton" className="pull-right btn btn-default" label="edge Delete Button"
                          onClick={handleEdgeDeleteButton()}>Delete Edge</FlatButton>
        </MuiThemeProvider>
  ))
    .add('material-ui raised button', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(darkBaseTheme)}>
            <RaisedButton key="edgeDeleteButton" className="pull-right btn btn-default" label="edgeDeleteButton"
                          onClick={action("Raised button clicked")}>Delete Edge</RaisedButton>
        </MuiThemeProvider>
    ))
    .add('material-ui Toggle switch', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(darkBaseTheme)}>
            <Toggle key="edgeToggleSwitch" className="pull-right btn btn-default" label="Edge Toggle"
                          onToggle={action("Toggle switch")}/>
        </MuiThemeProvider>
    ));

