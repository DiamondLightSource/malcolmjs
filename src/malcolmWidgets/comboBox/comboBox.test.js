import React from 'react';
import { createShallow, createMount } from 'material-ui/test-utils';
import WidgetComboBox from './comboBox.component';

describe('WidgetComboBox', () => {
  let shallow;
  let mount;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <WidgetComboBox
        Value="2"
        Pending={false}
        Choices={['1', '2', '3']}
        selectEventHandler={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  /*
  it('calls check', () => {
    const checkAction = jest.fn();
    const wrapper = mount(
      <WidgetCheckbox
        CheckState={false}
        checkEventHandler={() => {
          checkAction();
        }}
        Label="TestBox"
      />
    );

    wrapper.find('input').simulate('change');

    expect(checkAction.mock.calls.length).toEqual(1);
  });
  */
});
