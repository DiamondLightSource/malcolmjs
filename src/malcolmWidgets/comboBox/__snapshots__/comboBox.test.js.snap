// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetComboBox renders correctly 1`] = `
<FormControl
  className="WidgetComboBox-formControl-2"
  classes={
    Object {
      "fullWidth": "MuiFormControl-fullWidth-7",
      "marginDense": "MuiFormControl-marginDense-6",
      "marginNormal": "MuiFormControl-marginNormal-5",
      "root": "MuiFormControl-root-4",
    }
  }
  component="div"
  disabled={false}
  error={false}
  fullWidth={true}
  margin="none"
  required={false}
>
  <WithStyles(Select)
    className="WidgetComboBox-select-3"
    native={true}
    onChange={[Function]}
    value="2"
  >
    <option
      disabled={true}
      value={null}
    >
       
    </option>
    <option
      key="0"
      value="1"
    >
      1
    </option>
    <option
      key="1"
      value="2"
    >
      2
    </option>
    <option
      key="2"
      value="3"
    >
      3
    </option>
  </WithStyles(Select)>
</FormControl>
`;
